src/DependencyInjection/Compiler/TweakCompilerPass.php

```diff
@@ -99,4 +83,31 @@ public function applyContext(ContainerBuilder $container)
             }
         }
     }
+
+    /**
+     * Replaces the empty service name with the service id.
+     */
+    private function replaceBlockName(ContainerBuilder $container, Definition $definition, $id)
+    {
+        $arguments = $definition->getArguments();
+
+        // Replace empty block id with service id
+        if (empty($arguments) || 0 == \strlen($arguments[0])) {
+            // NEXT_MAJOR: Remove the condition when Symfony 2.8 support will be dropped.
+            if (method_exists($definition, 'setArgument')) {
+                $definition->setArgument(0, $id);
+            } else {
+                $definition->replaceArgument(0, $id);
+            }
+        } elseif ($id != $arguments[0] && 0 !== strpos(
+            $container->getParameterBag()->resolveValue($definition->getClass()),
+            'Sonata\\BlockBundle\\Block\\Service\\'
+        )) {
+            // NEXT_MAJOR: Remove deprecation notice
+            @trigger_error(
+                sprintf('Using service id %s different from block id %s is deprecated since 3.3 and will be removed in 4.0.', $id, $arguments[0]),
+                E_USER_DEPRECATED
+            );
+        }
+    }
```

Comment:
How about this?
```php
<?php
$arguments = $definition->getArguments();
// Replace empty block id with service id
if (empty($arguments) || 0 == \strlen($arguments[0])) {
    // NEXT_MAJOR: Remove the if block when Symfony 2.8 support will be dropped.
    if (method_exists($definition, 'setArgument')) {
        $definition->setArgument(0, $id);
        return;
    }
    $definition->replaceArgument(0, $id);
    return;
}
if ($id != $arguments[0] && 0 !== strpos(
    $container->getParameterBag()->resolveValue($definition->getClass()),
    'Sonata\\BlockBundle\\Block\\Service\\'
)) {
    // NEXT_MAJOR: Remove deprecation notice
    @trigger_error(
        sprintf('Using service id %s different from block id %s is deprecated since 3.3 and will be removed in 4.0.', $id, $arguments[0]),
        E_USER_DEPRECATED
    );
}
```